### PR TITLE
apply_change_to_project: Handle remove configuration list change.

### PR DESCRIPTION
**Pull request by byohay**
Also, refactor applying added changes for symmetry with removed changes.

### CODE REVIEW

File: lib/kintsugi/apply_change_to_project.rb [hunk 1]
```diff
@@ -210,23 +210,30 @@ def apply_change_to_component(parent_component, change_name, change, parent_chan
...
-      (change[:removed] || []).each do |removed_change|
-        child =
-          child_component_of_object_list(component, removed_change, removed_change["displayName"])
-        remove_component(child, removed_change)
+      if change[:removed].is_a?(Hash)
+        remove_component(component, change[:removed])
+      elsif change[:removed].is_a?(Array)
+        (change[:removed]).each do |removed_change|
+          child =
+            child_component_of_object_list(component, removed_change, removed_change["displayName"])
```
Comment (ashdnazg) [inline lib/kintsugi/apply_change_to_project.rb:220]: this function expects component to be non-nil

File: lib/kintsugi/apply_change_to_project.rb [hunk 2]
```diff
@@ -210,23 +210,30 @@ def apply_change_to_component(parent_component, change_name, change, parent_chan
...
+      elsif !change[:removed].nil?
+        raise MergeError, "Unsupported change type for #{change[:removed]}"
       end
 
-      (change[:added] || []).each do |added_change|
-        is_object_list = component.is_a?(Xcodeproj::Project::ObjectList)
-        add_child_to_component(is_object_list ? parent_component : component, added_change,
-                               change_path)
+      if change[:added].is_a?(Hash)
+        add_child_to_component(parent_component, change[:added], change_path)
```
Comment (ashdnazg) [inline lib/kintsugi/apply_change_to_project.rb:228]: same

**Review by ashdnazg — CHANGES_REQUESTED** [review]
F&S

**Comment by byohay** [comment]
@ashdnazg I added guards around places that need `component`. I also added this dependency in change[:added], as it always needs the parent component.

**Review by ashdnazg — APPROVED** [review]
I think it's ok,
Note that in the commit message you still mention symmetry, but that's no longer the case.
Also we need to remember to fail `added` if it will override a different existing value. Probably in a different PR.

**Pull request merged**